src/QueryBuilder/AbstractDMLQueryBuilder.php

```diff
@@ -174,23 +186,29 @@ final protected function prepareTraversable(Traversable $rows): Iterator|array
      * @param string $table The table name.
      * @param iterable $rows The rows to be batch inserted into the table.
      * @param string[] $columnNames The column names.
-     * @param array $params The binding parameters that will be generated by this method.
+     * @param int $rowsAtOnceLimit The limit of rows inserted at once.
      *
-     * @return string[] The values.
+     * @return QueryStatementParameters[] The values.
      *
      * @psalm-param ParamsType $params
      */
-    protected function prepareBatchInsertValues(string $table, iterable $rows, array $columnNames, array &$params): array
+    protected function prepareBatchInsertValues(string $table, iterable $rows, array $columnNames, int $rowsAtOnceLimit = 0): array
     {
-        $values = [];
+        $queryStatementParameters = [];
         /** @var string[] $names */
         $names = array_values($columnNames);
         $keys = array_fill_keys($names, false);
         $columns = $this->schema->getTableSchema($table)?->getColumns() ?? [];
 
+        $statementParameters = new QueryStatementParameters();
+        $maxParametersLimit = $this->queryBuilder->getParametersLimit();
```

Comment:
```suggestion
        $maxParametersLimit = $this->getParametersLimit();
```

Let's do it protected inside `AbstractDMLQueryBuilder`